_posts/2023-01-23-install-jekyll-and-bundler.md

```diff
@@ -0,0 +1,133 @@
+
+- Optionally, you can export the binary to use it with the host directly without requiring the `apx run` command using the following command:-
+
+```bash
+apx export --bin jekyll
+apx export --bin bundler
+```
+
+### Fedora container
+
+- Enter the container using the following command:-
+
+```bash
+apx enter --dnf
+```
+- Install the required prerequisite packages using the following command:-
+
+```bash
+sudo dnf install ruby ruby-devel make gcc g++
+```
+
+- Now, Install Jekyll using the following command:-
+
+```bash
+gem install jekyll
+```
+
+- Now you have successfully installed the packages, You can execute them using `apx run --dnf "command"` or `apx run --dnf 'command'` or directly by entering the container using `apx enter --dnf`.
+
+- Optionally, you can export the binary to use it with the host directly without requiring the `apx run --dnf` command using the following command:-
+
+```bash
+apx export --dnf --bin jekyll
+apx export --dnf --bin bundler
+```
+
+### Arch Linux container
+
+- Enter the container using the following command:-
+
+```bash
+apx enter --aur
+```
+- Install the required prerequisite packages using the following command:-
+
+```bash
+sudo pacman -S ruby base-devel
+```
+
+- Now, Install Jekyll using the following command:-
+
+```bash
+gem install jekyll
+```
+
+- Now you have successfully installed the packages, You can execute them using the `apx run --aur "command"` or `apx run --aur 'command'` or directly by entering the container using `apx enter --aur`.
+
+- Optionally, you can export the binary to use it with the host directly without requiring the `apx run --aur` command using the following command:-
+
+```bash
+apx export --aur --bin jekyll
+apx export --aur --bin bundler
+```
+
+## Testing sites from a repository locally
+
+- Clone a repository using `git` or `gh`.
+- Move to the destination of the cloned directory using `cd <path/to/directory>`.
+- Run `bundler install` to install necessary packages (this step is required only once and doesn't require re-running it in future, and you must have a Gemfile in the repository for this to work).
+- Run `bundler exec jekyll build` to build the page to `./_site` once. Then you can either test the pages manually or use the `bundler exec jekyll serve` command.
+- Run `bundler exec jekyll serve` to build your site any time a source file changes and serve it locally.
+  - Navigate to `http://127.0.0.1:4000/` or `http://localhost:4000/` in your browser to preview and test the page.
+- Now, commit the changes using `git` and create a PR in GitHub.
+
+**_Tip_**:-
+
+- You can test your pages on your phone using `bundler exec jekyll serve --host=<ip>`. 
+- Using `0.0.0.0` instead of a specific IP binds port 4000 to any interface, which is prone to be blocked by your routers firewalls, so we recommended using a particular IP address with the `--host` flag. Now, after executing the command in any browser on your phone, go to this address `<ip>:4000`.
```

Comment:
```suggestion
- Using `0.0.0.0` instead of a specific IP binds port 4000 to any interface, which is prone to be blocked by your routers firewalls. That's why we recommended using a particular IP address with the `--host` flag. After executing the command in any browser on your phone, go to this address `<ip>:4000`.
```